File: 프로그래머스/2/148653. 마법의 엘리베이터/마법의 엘리베이터.java
class Solution {
    public int solution(int storey) {
        
        int cnt = 0;

        while(storey != 0){
            
            int left = storey % 10;
            storey /= 10;
            
            if(left == 0) continue;
            
            if(left > 5){
                storey ++;
                cnt += 10 - left;
            }else if(left < 5){
                cnt += left;
            }else{
                if(storey % 10 >= 5){
                    storey ++;
                }
                cnt += left;
            }
        }
        
        return cnt;
    }
}

File: 프로그래머스/2/148653. 마법의 엘리베이터/마법의 엘리베이터.js
function solution(storey) {
    
    var answer = 0;
    
    while(storey != 0){
        var left = storey % 10;
        storey = Math.floor(storey/10);
        if(left > 5){
            storey ++;
            answer += 10 - left;
        }else if(left < 5){
            answer += left;
        }else{
            if(storey % 10 >= 5){
                storey++;
            }
            answer += left;
        }
    }
    
    return answer;
}